spec/unit/parser_spec.rb

```diff
@@ -0,0 +1,255 @@
+require 'spec_helper'
+require 'iknow_params/parser'
+
+RSpec.describe IknowParams::Parser do
+  class Controller
+    attr_reader :params
+    def initialize(params)
+      @params = params
+    end
+  end
+
+  before { Controller.include IknowParams::Parser }
+
+  describe "#parse_param" do
+
+    let(:parsed) { Controller.new(@params).parse_param(@param, with: serializer) }
+
+    context "no serializer specified" do
+      let(:serializer) { nil }
+
+      it "returns the value" do
+        @params = {name: "foo"}
```

Comment:
Is it important that each example is different? What's gained by having `@params` be `{id: "10"}` in one example and `{id: "5"}` in another?  Would it be cleaner to instead set up  context groups for equivalent situations like `context "with valid params" do` and `context "with invalid params" do`?